web/pgadmin/tools/erd/static/js/erd_tool/components/MainToolBar.jsx

```diff
@@ -134,6 +138,15 @@ export function MainToolBar({preferences, eventBus, fillColor, textColor, notati
         isDirtyRef.current = isDirty;
         setDisableButton('save', !isDirty);
       }],
+      [ERD_EVENTS.SAVE_ERD_TOOL_DATA, (data)=>{
+        const save_app_state = preferencesStore?.getPreferencesForModule('misc')?.save_app_state;
+        if(save_app_state){
+          saveToolData({'tool_name': 'ERD',
```

Comment:
saveToolData should be arg based. Eg. saveToolData('ERD', connectionInfo.trans_id, otherData). Internally you can form the JSON like required by API.

Reply:
Done.